rgdps/api/gd/user_relationships.py

```diff
@@ -0,0 +1,282 @@
+        logger.info(
+            f"Failed to delete friend request from/to {target_id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    logger.info(f"{user} successfully deleted friend request from/to {target_id}.")
+    return responses.success()
+
+
+async def friend_request_accept(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    target_id: int = Form(..., alias="targetAccountID"),
+    request_id: int = Form(..., alias="requestID"),
+):
+    result = await friend_requests.accept(
+        ctx,
+        sender_user_id=target_id,
+        recipient_user_id=user.id,
+        request_id=request_id,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to accept friend request from {target_id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    logger.info(f"{user} successfully accepted friend request from {target_id}.")
+    return responses.success()
+
+
+async def user_relatoionship_list_get(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    relationship_type: UserRelationshipType = Form(..., alias="type"),
+):
+
+    result = await user_relationships.get_user(
+        ctx,
+        user.id,
+        relationship_type,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to view user relationship of {user.id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    if not result.relationships:  # No relationships found.
+        return responses.code(-2)
+
+    response = "|".join(
+        gd_obj.dumps(
+            [
+                gd_obj.create_profile(relationship.user),
+                gd_obj.create_user_relationship_list(relationship.relationship),
+            ],
+        )
+        for relationship in result.relationships
+    )
+
+    # Mark them as seen.
+    await user_relationships.mark_all_as_seen(ctx, user.id, relationship_type)
+    logger.info(f"{user} successfully viewed user relationship.")
+
+    return response
+
+
+async def user_friend_remove(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    target_id: int = Form(..., alias="targetAccountID"),
+):
+    result = await user_relationships.remove_friendship(
+        ctx,
+        user.id,
+        target_id,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to remove friend {target_id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    logger.info(f"{user} successfully removed friend {target_id}.")
+    return responses.success()
+
+
+async def block_user(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    target_id: int = Form(..., alias="targetAccountID"),
+):
+
+    # Remove friendship if exists.
+    friendship = await user_relationships.get_user(
```

Comment:
This logic should be part of a block user usecase. An endpoint should only call one usecase and only once.

Reply:
resolved in eccb259